src/features.rs

```diff
@@ -273,23 +270,20 @@ impl DenoiseFeatures {
 ///
 /// The Fourier transform goes in `x` and the band energies go in `ex`.
 fn transform_input(
-    fft: &mut crate::fft::RealFft,
     input: &[f32],
     lag: usize,
     window_buf: &mut [f32; WINDOW_SIZE],
-    x: &mut [Complex],
+    x: &mut DynRealDft<f32>,
     ex: &mut [f32],
 ) {
     let input = &input[input.len().checked_sub(WINDOW_SIZE + lag).unwrap()..];
     crate::apply_window(&mut window_buf[..], input);
-    fft.forward(window_buf, x);
+    *x = window_buf.real_fft();
```

Comment:
This will allocate, right?

Reply:
Unfortunately, I think yes. The definition for `DynRealDft` contains a `Box`:
```
pub struct DynRealDft<T> {
    original_length: usize,
    inner: Box<[Complex<T>]>,
}
```

So each time a `DynRealDft` is created, we allocate. The const sized equivalent in the `easyfft` library does not allocate:
```
pub struct RealDft<T, const SIZE: usize>
where
    [T; SIZE / 2 + 1]: Sized,
{
    inner: [Complex<T>; SIZE / 2 + 1],
}
```
`[nnnoiseless](https://github.com/jneem/nnnoiseless)` is ideal for using this, except that it depends on unstable features in rustc.